projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts

```diff
@@ -75,13 +75,21 @@ export class NgxIndexedDBService {
    * @Return the current version of database as number
    */
   getDatabaseVersion(): Observable<number | string> {
-    return new Observable((obs) => {
+    return new Observable((obs: Subscriber<number | string>) => {
+      let db: IDBDatabase | null = null;
       openDatabase(this.indexedDB, this.dbConfig.name, this.dbConfig.version)
-        .then((db: IDBDatabase) => {
-          obs.next(db.version);
+        .then((database: IDBDatabase) => {
+          db = database;
+          obs.next(database.version);
           obs.complete();
         })
-        .catch((err) => obs.error(`error during get version of database => ${err} `));
+        .catch((err) => obs.error(`Error during get version of database => ${err}`))
+        .finally(() => {
+          if (db) {
+            db.close();
```

Comment:
The database connection is not closed at any point? Can you check if the connection is actually closed after completing the operation? For example: I execute the add method the connection is closed after completing this method?

Reply:
i performed getDatabaseVersion removing method to close

getDatabaseVersion(): Observable<number | string> {
    return new Observable((obs: Subscriber<number | string>) => {
      let db: IDBDatabase | null = null;
      openDatabase(this.indexedDB, this.dbConfig.name, this.dbConfig.version)
        .then((database: IDBDatabase) => {
          db = database;
          obs.next(database.version);
          obs.complete();
          this.checkConnectionState(db);
        })
        .catch((err) => obs.error(`Error during get version of database => ${err}`))
        .finally(() => {
          // if (db) {
          //   db.close();
          // }
          // console.log('db closed');
        });
    });
  }
  
  checkConnectionState(db: IDBDatabase | null): void {
    try {
      db.transaction(['test'], 'readonly');
      console.log('Database connection is open.');
    } catch (e) {
      console.error('Error:', e);
    }
  }
  
  in this method i check in console if connection is still open and as a result i see 'Database connection is open'
  
  i do the same test with add and got same result
  
  I hope i answer your question